enzy_htp/structure/structure_region/capping.py

```diff
@@ -50,124 +62,222 @@ def capping_with_residue_terminals(raw_region,
 
-class ResidueCap(Residue):
-    """the residue cap that link to the corresponding
+class ResidueCap(Residue, ABC):
+    """The residue cap that link to the corresponding
     Residue while not having the children atoms actually in
-    that Residue.
+    that Residue. It ensures that given the defined terminal atom, it can 
+    generate a realistic geometry.
+
     This is mainly used in handling cases like capping atoms that
-    does not exists in the real structure.
-    Attribute:
-        link_residue
-        children/atoms
-        parent = None
-        link_atom:
-            the linking atom in original res
-        socket_atom:
-            the socket from original stru to pluging the cap.
-            the cap is replacing the atom.
-        plug_atom:
-            the pluging in atom in the cap
+    do not exist in the real structure.
+
+    Attributes:
+        link_residue: The Residue() being capped.
+        link_atom: THe terminal Atom() that needs capping specifically.
+        socket_atom: Atom() that connects to the link_atom in the normal Structure().
+        _terminal_type: What type of terminal is it? n-term, c-term, etc.
+        atoms: A List[Atom] of the Atom()'s in the ResidueCap.
     
-    Property:
-        cap_type"""
+    Properties:
+        idx: The idx of the Residue() being capped.
+        chain: The chain name of the Residue() being capped.
+        name: The residue name of the Residue() being capped.
+        term_type: The property getter of _terminal_type
+        net_charge: The net charge of the ResidueCap() chemical group.
+        """
     def __init__(self, link_residue: Residue,
                  link_atom: Atom, socket_atom: Atom,
-                 atoms: List[Atom], plug_atom: Atom):
+                    terminal_type:str):
+        """Simplistic ctor. Will throw if link_atom is NOT in link_residue."""
         self.link_residue = link_residue
         self.link_atom = link_atom
         self.socket_atom = socket_atom
         if link_atom not in link_residue.atoms:
             _LOGGER.error("link atom must be one of the link residue atoms")
             raise ValueError
-        self.atoms = atoms
-        self.plug_atom = plug_atom
-        if plug_atom not in self.atoms:
-            _LOGGER.error("plug atom must be one of the atoms")
-            raise ValueError
 
-        self._idx = None
-        self._name = None
         self._rtype = chem.ResidueType.RESIDUECAP
+        self._terminal_type = terminal_type 
+
+        if self.term_type == 'nterm':
+            self.atoms = self.get_nterm_atoms()
+        elif self.term_type == 'cterm':
+            self.atoms = self.get_cterm_atoms()
+
         self.set_ghost_parent()
-        self.set_children(atoms)
-    
+
+        self.anchor()
+
+
+    @abstractmethod
+    def anchor(self) -> None:
+        """Function that moves the ResidueCap()'s Atom()'s to be in the correct position for the type of Cap that it is."""
```

Comment:
This desc is a bit vague I think. especially the "for the type of Cap that it is." part

Reply:
I agree this is vague. I have made it more specific/detailed now.